packages/connection-form/src/components/advanced-options-tabs/csfle-tab/csfle-tab.tsx

```diff
@@ -0,0 +1,188 @@
+  noTLS?: boolean;
+  clientCertIsOptional?: boolean;
+}
+
+const options: KMSProviderMetadata[] = [
+  {
+    title: 'Local KMS',
+    kmsProvider: 'local',
+    noTLS: true,
+  },
+  {
+    title: 'AWS',
+    kmsProvider: 'aws',
+  },
+  {
+    title: 'GCP',
+    kmsProvider: 'gcp',
+  },
+  {
+    title: 'Azure',
+    kmsProvider: 'azure',
+  },
+  {
+    title: 'KMIP',
+    kmsProvider: 'kmip',
+    clientCertIsOptional: false,
+  },
+];
+
+const containerStyles = css({
+  marginTop: spacing[3],
+});
+
+function CSFLETab({
+  connectionOptions,
+  updateConnectionFormField,
+  errors: errors_,
+}: {
+  errors: ConnectionFormError[];
+  updateConnectionFormField: UpdateConnectionFormField;
+  connectionOptions: ConnectionOptions;
+}): React.ReactElement {
+  const autoEncryptionOptions =
+    connectionOptions.fleOptions?.autoEncryption ?? {};
+
+  const errors = errorsByFieldTab(errors_, 'csfle');
+
+  const handleFieldChanged = useCallback(
+    (
+      key: keyof AutoEncryptionOptions,
+      value?: AutoEncryptionOptions[keyof AutoEncryptionOptions]
+    ) => {
+      return updateConnectionFormField({
+        type: 'update-csfle-param',
+        key: key,
+        value,
+      });
+    },
+    [updateConnectionFormField]
+  );
+
+  return (
+    <div className={containerStyles}>
+      <Banner>
+        Client-side Field-Level Encryption is an Enterprise/Atlas-only feature
+        of MongoDB.&nbsp;
+        <Link href="https://www.mongodb.com/docs/drivers/security/client-side-field-level-encryption-guide/">
+          Learn More
+        </Link>
+      </Banner>
+      <FormFieldContainer>
+        <TextInput
+          onChange={({ target: { value } }: ChangeEvent<HTMLInputElement>) => {
+            handleFieldChanged('keyVaultNamespace', value);
+          }}
+          name="csfle-keyvault"
+          data-testid="csfle-keyvault"
+          label="Key Vault Namespace"
+          type="text"
+          optional={false}
+          value={autoEncryptionOptions.keyVaultNamespace || ''}
+          errorMessage={errorMessageByFieldName(errors, 'keyVaultNamespace')}
+          state={
+            fieldNameHasError(errors, 'keyVaultNamespace') ? 'error' : 'none'
+          }
+          spellCheck={false}
+          description="Specify a collection in which data encryption keys are stored in the format <db>.<collection>."
+        />
+      </FormFieldContainer>
+      <EncryptedFieldConfigInput
+        // TODO(COMPASS-5645): This says 'schemaMap', which is the
+        // FLE1 equivalent of the FLE2 'encryptedFieldConfig[Map?]'.
+        // Once 'encryptedFieldConfig' is available, we will start
+        // using it instead.
+        encryptedFieldConfig={autoEncryptionOptions?.schemaMap}
+        errorMessage={errorMessageByFieldName(errors, 'schemaMap')}
+        onChange={(value: Document | undefined) => {
+          handleFieldChanged('schemaMap', value);
+        }}
+      />
+      <div className={withMarginStyles}>
```

Comment:
Same as above, how would you feel about wrapping or replacing this with the `FormFieldContainer` so that it gets a little bit of top margin? We might be able to remove the `withMarginStyles` then.